WordPress/src/main/java/org/wordpress/android/ui/mysite/jetpackbadge/JetpackPoweredBottomSheetFragment.kt

```diff
@@ -0,0 +1,121 @@
+    @Inject lateinit var adapter: JetpackPoweredAdapter
+    @Inject lateinit var activityLauncherWrapper: ActivityLauncherWrapper
+    private val viewModel: JetpackPoweredDialogViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.jetpack_powered_bottom_sheet, container)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with(JetpackPoweredBottomSheetBinding.bind(view)) {
+            initializeViews()
+            setupObservers()
+        }
+    }
+
+    private fun JetpackPoweredBottomSheetBinding.initializeViews() {
+        contentRecyclerView.layoutManager = LinearLayoutManager(requireActivity())
+        contentRecyclerView.adapter = adapter
+        contentRecyclerView.disableAnimation()
+
+        primaryButton.setOnClickListener { viewModel.openJetpackAppDownloadLink() }
+
+        val fullScreen = arguments?.getBoolean(KEY_FULL_SCREEN, false) ?: false
+
+        if (fullScreen) {
+            secondaryButton.visibility = View.VISIBLE
+            secondaryButton.text = getString(R.string.wp_jetpack_continue_to_reader)
+            secondaryButton.setOnClickListener { viewModel.dismissBottomSheet() }
+        }
+
+        (dialog as? BottomSheetDialog)?.apply {
+            setOnShowListener {
+                val bottomSheet: FrameLayout = dialog?.findViewById(
+                        com.google.android.material.R.id.design_bottom_sheet
+                ) ?: return@setOnShowListener
+                val bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet)
+                if (fullScreen && bottomSheet.layoutParams != null) {
+                    showFullScreenBottomSheet(bottomSheet)
+                }
+                expandBottomSheet(bottomSheetBehavior)
+            }
+        }
+    }
+
+    private fun JetpackPoweredBottomSheetBinding.setupObservers() {
+        viewModel.start()
+
+        viewModel.uiState.observe(this@JetpackPoweredBottomSheetFragment) { uiState ->
+            adapter.submitList(uiState?.uiItems ?: listOf())
+        }
+
+        viewModel.action.observe(viewLifecycleOwner) { action ->
+            when (action) {
+                is OpenPlayStore -> {
+                    dismiss()
+                    activity?.let {
+                       activityLauncherWrapper.openPlayStoreLink(it, JETPACK_PACKAGE_NAME)
+                    }
+                }
+                is DismissDialog -> {
+                    dismiss()
+                }
+            }.exhaustive
+        }
+    }
+
+    private fun showFullScreenBottomSheet(bottomSheet: FrameLayout) {
+        val layoutParams = bottomSheet.layoutParams
+        layoutParams.height = Resources.getSystem().displayMetrics.heightPixels
+        bottomSheet.layoutParams = layoutParams
```

Comment:
Could this have been just:

```suggestion
bottomSheet.layoutParams.height = Resources.getSystem().displayMetrics.heightPixels
```

or am I missing something?